tutorials/timefreq/plot_MorletWavelet.py

```diff
@@ -0,0 +1,247 @@
+#
+
+###################################################################################################
+#
+# We can also change the frequencies passed to the morlet-wavelet transform algorithm.
+# For example, let's use an array of frequencies from 15 Hz to 50 Hz with a spacing of 5 Hz.
+
+###################################################################################################
+
+# Settings for the wavelet transform algorithm
+freqs = np.arange(15, 50, 5)
+
+# Compute wavelet transform using compute morlet wavelet transform algorithm
+mwt = compute_wavelet_transform(sig, fs=fs, n_cycles=7, freqs=freqs)
+
+###################################################################################################
+
+# Plot morlet wavelet transform
+fig, ax = plt.subplots()
+ax.imshow(abs(mwt), aspect='auto')
+ax.invert_yaxis()
+ax.set_xlabel('time (s)')
+ax.set_xticks(np.linspace(0, times.size, 5))
+ax.set_xticklabels(np.round(np.linspace(times[0], times[-1], 5), 2))
+ax.set_ylabel('freq (Hz)')
+ax.set_yticks(np.linspace(0, freqs.size, 5))
+ax.set_yticklabels(np.round(np.linspace(freqs[0], freqs[-1], 5), 2))
+fig.show()
+
+###################################################################################################
+#
+# From the plot above, you can see the Morlet-wavelet transformed signal for the new frequency range.
+#
+
+###################################################################################################
+#
+# It is useful to describe the parameters needed to define a Morlet wavelet. These parameters are
+# the sampling frequency, the fundamental frequency, and the number of cycles per frequency.
+#
+# It is also usefule to describe what the Morlet wavelet is in the context of this tutorial.
+# Simply put, a morlet wavelet takes a raw input signal and multiplies it by a Gaussian envelope.
+# It is useful in cases where a signal's amplitude varies over time.
+#
+# For more information on Morlet wavelets, see:
+# Mike X Cohen, 2019, "Morlet wavelets in time and frequency,"
+# YouTube, https://www.youtube.com/watch?v=7ahrcB5HL0k.
+#
+
+###################################################################################################
+
+# Example plot of morlet wavelet
+# ------------------------------
+#
+# Here, I provide an example plot of a Morlet wavelet to demonstrate my previous description
+
+###################################################################################################
+
+# Use the scipy.signal function 'morlet' to create a wavelet.
+# Here, we use a length of 175 with a total of 7 cycles.
+
+# Define sampling rate, number of cycles, fundamental frequency, and length for the wavelet.
+fs = 500
+n_cycles = 7
+freq = 5
+s = 1.0
+w = n_cycles
+M = int(n_cycles * fs / freq)
+
+# Create wavelet
+wavelet = signal.morlet(M, w, s)
+
+# Plot wavelet
+fig = plt.figure()
+ax = fig.add_subplot(111, projection='3d')
+ax.plot(np.linspace(0, s, wavelet.size), wavelet.real, wavelet.imag)
+fig.show()
```

Comment:
Axes could be labeled here. It may also be helpful to remind readers about convolution somewhere.

Reply:
also, `fig.show()` isn't needed